testnet/stacks-node/src/event_dispatcher.rs

```diff
@@ -25,9 +28,59 @@ const STATUS_RESP_TRUE: &str = "success";
 const STATUS_RESP_NOT_COMMITTED: &str = "abort_by_response";
 const STATUS_RESP_POST_CONDITION: &str  = "abort_by_post_condition";
 
+pub const PATH_MEMPOOL_TX_SUBMIT: &str = "new_mempool_tx";
+pub const PATH_BLOCK_PROCESSED: &str = "new_block";
+
 impl EventObserver {
 
-    fn send(&mut self, filtered_events: Vec<&(bool, Txid, &StacksTransactionEvent)>, chain_tip: &ChainTip) {
+    fn send_payload(&self, payload: &serde_json::Value, path: &str) {
+        let endpoint = format!("{}{}{}",
+                               &self.endpoint,
+                               if self.endpoint.ends_with("/") { "" } else { "/" },
+                               path);
+
+        let mut backoff: f64 = 1.0;
+        let mut rng = thread_rng();
+        loop {
+            let response = Client::new()
+                .post(&endpoint)
+                .json(payload)
+                .send();
+
+            match response {
+                Ok(response) => {
+                    if response.status().is_success() {
+                        break;
+                    }
+                    error!("Event dispatcher: POST {} failed with error {:?}", self.endpoint, response);
+                },
+                Err(e) => {
+                    error!("Event dispatcher: POST {} failed with error {:?}", self.endpoint, e);
+                }
+            };
+
+            backoff = (2.0 * backoff + (backoff * rng.gen_range(0.0, 1.0))).min(60.0);
+            let duration = Duration::from_millis((backoff * 1_000.0) as u64);
+            info!("Event dispatcher will retry posting in {:?}", duration);
+            sleep(duration);
+        };
+    }
+
+    fn make_new_mempool_txs_payload(transactions: Vec<StacksTransaction>) -> serde_json::Value {
+        let raw_txs = transactions.into_iter().map(|tx| {
+            serde_json::Value::String(
+                format!("0x{}", &bytes_to_hex(&tx.serialize_to_vec())))
+        }).collect();
+
+        serde_json::Value::Array(raw_txs)
+    }
+
+    fn send_new_mempool_txs(&self, payload: &serde_json::Value) {
+        self.send_payload(payload, PATH_MEMPOOL_TX_SUBMIT);
```

Comment:
Unclear to me, which thread is potentially blocked, if the `/new_mempool_tx` is not being responsive?
Should `send_payload` be conditionally blocking, or is this a desired behavior?

Reply:
It's the block processing thread (the same thread that blocks on `/new_block`).

I think it's fine for this to be non-blocking -- though that's more of a question for the sidecar interface: is it okay for a `/new_mempool_tx` request to be received for a transaction _after_ that transaction was included in a block (that's an extreme example of a consequence of not-blocking here). It's probably preferable from the stacks-node perspective that this be non-blocking (I think it would require a new "mempool event sender" thread, though).

Reply:
Regarding this -- I think it's okay to keep this blocking for now, since it will only impact nodes that use a side car, but we should start an issue to revisit.

Reply:
Note that the sidecar's mempool POST handler and block POST handler will use the same db update queue. So even if this endpoint is made async, if something is really slow in the sidecar, then the node will end up waiting behind a subsequent block POST anyway.

Reply:
If we're worried about backpressure from high transaction volume, maybe the Stacks node can just timeout transaction event postings if stalling is going to be a problem?  I'd prefer dropping transactions to implementing a separate transaction-posting thread, since transaction state is ultimately soft state in the face of block data (which _must_ be transmitted in a blocking fashion).